### PR TITLE
ADD psqlodbc test for sys.image

### PR DESCRIPTION
**Pull request by HackonToby**
Signed-off-by: Toby Cao <tobyc@bitquilltech.com>

### Description

Add test case for sys.image

Table Creation
Insertion_Success
Update_Success
View Creation
String_Operator(Disabled)
 
### Issues Resolved

BABELFISH-567

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/odbc/psqlodbc/test/image.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
+#include "../conversion_functions_common.h"
+#include "../psqlodbc_tests_common.h"
+#include <fstream>
+
+const string TABLE_NAME = "master_dbo.image_table_odbc_test";
+const string COL1_NAME = "pk";
+const string COL2_NAME = "data";
+const string DATATYPE_NAME = "sys.image";
+const string VIEW_NAME = "master_dbo.image_view_odbc_test";
```
Comment (ColinKYuen) [inline test/odbc/psqlodbc/test/image.cpp:15]: nit: could we add this beside the table name

File: test/odbc/psqlodbc/test/image.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+// Relative to where `./build/main` is ran
+const std::string INSERTED_FILE_PATH = "psqlodbc/Images/Black.png";
+const std::string UPDATED_FILE_PATH = "psqlodbc/Images/circle.png";
+static string INSERTED_IMAGE_HEX_STR;
+static string UPDATED_IMAGE_HEX_STR;
```
Comment (ColinKYuen) [inline test/odbc/psqlodbc/test/image.cpp:21]: Can these be const?

File: test/odbc/psqlodbc/test/image.cpp [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+#include "../psqlodbc_tests_common.h"
+#include <fstream>
+
+const string TABLE_NAME = "master_dbo.image_table_odbc_test";
+const string COL1_NAME = "pk";
+const string COL2_NAME = "data";
+const string DATATYPE_NAME = "sys.image";
+const string VIEW_NAME = "master_dbo.image_view_odbc_test";
+
+// Relative to where `./build/main` is ran
```
Comment (ColinKYuen) [inline test/odbc/psqlodbc/test/image.cpp]: My bad on this.
Should be relative to where the terminal's location when calling the executable.

File: test/odbc/psqlodbc/test/image.cpp [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  const vector<string> NAME_EXPECTED = {"int4", "unknown"};
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  testCommonColumnAttributes(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS.size(), COL1_NAME, LENGTH_EXPECTED, 
+    PRECISION_EXPECTED, SCALE_EXPECTED, NAME_EXPECTED);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Image, Insertion_Success) {
+  const vector<string> INSERTED_VALUES = {
```
Comment (ColinKYuen) [inline test/odbc/psqlodbc/test/image.cpp]: Could we change these to `EXPECTED_INSERTED_VALUES`, ditto to the rest below.

File: test/odbc/psqlodbc/test/image.cpp [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  insertValuesInTable(ServerType::PSQL, TABLE_NAME, INSERT_STRING, NUM_OF_DATA);
+  verifyValuesInObject(ServerType::PSQL, TABLE_NAME, COL1_NAME, INSERTED_VALUES, INSERTED_VALUES);
+
+  createView(ServerType::PSQL, VIEW_NAME, VIEW_QUERY);
+  verifyValuesInObject(ServerType::PSQL, VIEW_NAME, COL1_NAME, INSERTED_VALUES, INSERTED_VALUES);
+
+  dropObject(ServerType::PSQL, "VIEW", VIEW_NAME);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
```
Comment (faviansamatha) [inline test/odbc/psqlodbc/test/image.cpp:184]: Can we have a set of tests that uses operators like DATALENGTH and Substring? According to this: https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/t-sql/data-types/ntext-text-and-image-transact-sql.md, there's a list of functions and statements that could be used for the image datatype.  If we choose to pursue this, please first make sure that they are supported in BBF. If it isn't we could probably skip them.

**Review by ColinKYuen — COMMENTED** [review]
Minor nits, other than that tests are good.

File: test/odbc/psqlodbc/test/image.cpp [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  testCommonColumnAttributes(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS.size(), COL1_NAME, LENGTH_EXPECTED, 
+    PRECISION_EXPECTED, SCALE_EXPECTED, NAME_EXPECTED);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Image, Insertion_Success) {
+  const vector<string> INSERTED_VALUES = {
+    "0x" + INSERTED_IMAGE_HEX_STR
```
Comment (faviansamatha) [inline test/odbc/psqlodbc/test/image.cpp]: sys.images says they support `binary data`. Would inserting regular binary data work as well (like how you would insert for varbinary?). If so, let's add test cases for this as well.

This goes the same with update.

File: test/odbc/psqlodbc/test/image.cpp (hunk 2 above)
Comment (HackonToby) [reply test/odbc/psqlodbc/test/image.cpp:21]: += won't be able to handle const but static

File: test/odbc/psqlodbc/test/image.cpp [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+  const vector<string> NAME_EXPECTED = {"int4", "unknown"};
+
+  createTable(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS);
+  testCommonColumnAttributes(ServerType::PSQL, TABLE_NAME, TABLE_COLUMNS.size(), COL1_NAME, LENGTH_EXPECTED, 
+    PRECISION_EXPECTED, SCALE_EXPECTED, NAME_EXPECTED);
+  dropObject(ServerType::PSQL, "TABLE", TABLE_NAME);
+}
+
+TEST_F(PSQL_DataTypes_Image, Insertion_Success) {
+  vector<string> INSERTED_VALUES = {
```
Comment (faviansamatha) [inline test/odbc/psqlodbc/test/image.cpp]: This shouldn't be capitalized if it isn't a constant

File: test/odbc/psqlodbc/test/image.cpp [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+    "0",      // Min, different format
+    "46",     // Rand
+    "49",     // Rand
+    "02",     // Rand, different format
+    "268435455",  // 7 Bytes
+    "4294967295", // 8 Bytes - 1
+    "4294967295", // 8 Bytes
+    "4294967296" // 8 Bytes + 1
+  };
+  vector<string> EXPECTED_VALUES = getExpectedResults_VarBinary(INSERTED_VALUES);
```
Comment (faviansamatha) [inline test/odbc/psqlodbc/test/image.cpp]: This shouldn't be capitalized if it isn't a constant

File: test/odbc/psqlodbc/test/image.cpp [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+
+  INSERTED_VALUES.push_back(" DECODE('" + INSERTED_IMAGE_HEX_STR + "', 'hex')");
```
Comment (faviansamatha) [inline test/odbc/psqlodbc/test/image.cpp]: I'm wondering, would it make sense to actually just have this inside of the INSERTED_VALUE, and then Modify the expected value array? May be a bit cleaner. We should also maybe explain why we're doing this.

File: test/odbc/psqlodbc/test/image.cpp [hunk 10]
```diff
@@ -0,0 +1,217 @@
+#include "../conversion_functions_common.h"
+#include "../psqlodbc_tests_common.h"
+#include <fstream>
+
+const string BBF_TABLE_NAME = "master.dbo.char_table_odbc_test";
+// For BBF Connection
+//   Cannot prepend database name when creating/dropping view
+//   Must prepend database name when selecting from view
+const string BBF_VIEW_NAME = "dbo.char_view_odbc_test";
```
Comment (ColinKYuen) [inline test/odbc/psqlodbc/test/image.cpp]:
```suggestion
const string BBF_TABLE_NAME = "master.dbo.image_table_odbc_test";
// For BBF Connection
//   Cannot prepend database name when creating/dropping view
//   Must prepend database name when selecting from view
const string BBF_VIEW_NAME = "dbo.image_view_odbc_test";
```

**Review by faviansamatha — APPROVED** [review]

**Review by rossd-bitquill — APPROVED** [review]

**Pull request merged**